### PR TITLE
feat(SelectDropdown): Add checkmark to selected option

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Add checkmark to selected SelectDropdown options. Als fixed some centering stuff with the optional leading icons.

### PR Checklist

- [x] Related to designs: [here](https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=60-3&t=6ZuMn9rXXCg4FPsq-0)
- [x] Related to JIRA ticket: [gm-579]
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

* Go to the SelectDropdown page
* Choose an item from the SelectDropdown options
* See the checkmark next to the selected option
* See icons are more visually aligned

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/6065)  | [Portal Env](http://www.google.fr/ 'Named link title')   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.17.7-alpha.880229.0<br/>@codecademy/gamut-kit@0.6.404-alpha.880229.0<br/>@codecademy/styleguide@66.18.2-alpha.880229.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66300bdbb310b83158f4a282--gamut-preview.netlify.app>https://66300bdbb310b83158f4a282--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66300bdbb310b83158f4a282>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66311590889a57008550cd04--gamut-preview.netlify.app>https://66311590889a57008550cd04--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66311590889a57008550cd04>Deploy Logs</a></p>

File: packages/gamut/src/Form/SelectDropdown/elements.tsx [hunk 1]
```diff
@@ -209,51 +211,64 @@ export const CustomContainer = ({
...
+
 export const formatOptionLabel = ({
   label,
   icon: Icon,
   size,
   subtitle,
   rightLabel,
   disabled,
-}: ExtendedOption) => {
+}: any) => {
```
Comment (aresnik11) [inline packages/gamut/src/Form/SelectDropdown/elements.tsx:237]: does this have to be `any`?

File: packages/gamut/src/Form/SelectDropdown/types.tsx [hunk 2]
```diff
@@ -140,6 +135,10 @@ export type SizedIndicatorProps = DropdownIndicatorProps<
 > &
   InternalSelectProps;
 
+export type CustomSelectComponentProps<
+  T extends React.JSXElementConstructor<any>
```
Comment (aresnik11) [inline packages/gamut/src/Form/SelectDropdown/types.tsx:139]: same here, does this have to be `any`?

File: packages/styleguide/stories/Atoms/FormInputs/examples.tsx [hunk 3]
```diff
@@ -158,29 +158,30 @@ export const SelectDropdownOptionVariantsExample: React.FC = () => (
...
+          value="queen"
+          name="other-dropdown"
           options={[
             {
               options: [
                 {
                   label: 'king@chess.com',
                   subtitle: 'The King of Chess',
-                  value: 'xxx',
+                  value: 'king',
```
Comment (aresnik11) [inline packages/styleguide/stories/Atoms/FormInputs/examples.tsx:172]: can you update the codes thats shown below the select dropdown to match this change just for consistency?

Comment (dreamwasp) [reply packages/styleguide/stories/Atoms/FormInputs/examples.tsx:172]: yupp! will do

File: packages/gamut/src/Form/SelectDropdown/elements.tsx (hunk 1 above)
Comment (dreamwasp) [reply packages/gamut/src/Form/SelectDropdown/elements.tsx:237]: oh whoops, will change

File: packages/gamut/src/Form/SelectDropdown/types.tsx (hunk 2 above)
Comment (dreamwasp) [reply packages/gamut/src/Form/SelectDropdown/types.tsx:139]: good catch, will change

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.17.7-alpha.f20a84.0<br/>@codecademy/gamut-kit@0.6.404-alpha.f20a84.0<br/>@codecademy/styleguide@66.18.2-alpha.f20a84.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66315844f64c28260e9a4824--gamut-preview.netlify.app>https://66315844f64c28260e9a4824--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66315844f64c28260e9a4824>Deploy Logs</a></p>

**Review by aresnik11 — APPROVED** [review]
LGTM!

**Pull request merged**